### ISSUE TITLE
Implement SynthPoRep

### ISSUE THREAD

**Issue by rjan90** [filecoin-project/lotus#10984]
WIP tracking issue for SynthPoReps: https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0059.md

**User story**

- As a storage provider (SP) I want to leverage SynthethicPoReps to reduce the temporary data stored between PreCommit and ProveCommit from ~400GiB to ≤ 25GiB.

**Acceptence criteria (SP):**

- Can be enabled by setting `SynthPoRep=true` in the config.toml file
- All sector types sealed with SynthPoRep passes WindowPoSt/WinningPoSt

**Acceptence criteria (SaaS):**

**Techincal scope**

```[tasklist]
### Synthetic PoRep Development
- [ ] https://github.com/filecoin-project/filecoin-ffi/pull/400
- [ ] https://github.com/filecoin-project/rust-fil-proofs/pull/1701
- [ ] https://github.com/filecoin-project/rust-filecoin-proofs-api/pull/87
- [ ] https://github.com/filecoin-project/filecoin-ffi/pull/404
- [ ] https://github.com/filecoin-project/lotus/pull/10970
- [x] fvm: https://github.com/filecoin-project/ref-fvm/issues/1791
- [x] actor: https://github.com/filecoin-project/builtin-actors/issues/1314
- [x] ffi: update fvm deps:
```

We also need to introduce a network upgrade to introduce synthetic porep, thus we need to implement the barebone for nv21 in lotus to unblock testing.

```[tasklist]
### add barebone nv21
- [x] fvm: https://github.com/filecoin-project/ref-fvm/issues/1791
- [x] actor: v12
- [x] gst: https://github.com/filecoin-project/go-state-types/issues/190
- [ ] https://github.com/filecoin-project/lotus/issues/10993
- [x] Integrate actors v12 into Lotus using the pack.sh script
```

```[tasklist]
### Testing
- [x] Butterfly testing plan for SynthPoRep: https://www.notion.so/pl-strflt/SynthPoRep-Butterfly-network-testing-59e22e4e10a24521b50135571e67391f?pvs=4
- [ ] Calibration testing plan for SynthPoRep
```

```[tasklist]
### Deliverable
- [x] Include segment on SP feedback of early testing [Orjan/nemo]
- [ ] Blog post/media for explaining why all SPs want to use this [luca/irene/nemo]
- [ ] Network upgrade timeline
```

Current proposals for the network upgrade timeline can be seen here: https://github.com/filecoin-project/core-devs/discussions/140#discussioncomment-6052423

**Comment by jennijuju** [comment]
added fvm and actor development handoff issue & added a tasklist for `add barebone nv21
` - without it, we wont be able to test the synthetic porep work.

**Comment by jennijuju** [comment]
May I get an update on the status?

**Comment by jennijuju** [comment]
Latest update from @cryptonemo

https://github.com/filecoin-project/rust-fil-proofs/issues/1711#issuecomment-1623558947

**Comment by jennijuju** [comment]
Should we at least include ONE SaaS provider in the testing plan  (butterfly might be a good idea), unless we believe the regular SP pipeline would fully cover that too.

**Comment by jennijuju** [comment]
@snadrus once the actor and fvm code is ready, Andy will continue https://github.com/filecoin-project/lotus/pull/10970 in this sprint

**Comment by jennijuju** [comment]
We are hoping to have a deployable butterfly that accepts SythPoRep by the end of. the sprint (July 17th)

**Comment by jennijuju** [comment]
@rjan90 can I get a status update on this, how is testing going? are we close to landing PRs to nv integration branch?

**Comment by rjan90** [comment]
**Status update:**

- I would say that testing on the butterfly-network for SynthPoReps can be called completed 🟢: https://github.com/filecoin-project/lotus/pull/11057#issuecomment-1669003199
   - Testing verified deals on Butterfly has been hard, due to rkh-keys being hard to surface. That said SynthPoRep FIL+ sectors was tested to be successful on [a 2k-network](https://www.notion.so/pl-strflt/SynthPoRep-testing-on-local-devnet-3d90c1e2d8554be6a395c81f003828d7?pvs=4#9ba039c19ab344e5b9e1e6db6a6e9de2).
      - Want to confirm new FIL+ flow with DDO, so resurfacing these test-cases in the mid-september Butterfly-reset makes more sense.
   - Bug for extending SynthPoRep sectors has landed. https://github.com/filecoin-project/go-state-types/pull/200. Confirming this fix on a local devnet will be easier once SynthPoRep is being merged into `feat/nv21`
   
**Next steps:**
- Get reviews and land dependencies in their nv21 specific branches.
- Rebase the [Lotus introduce SynthPoRep PR](https://github.com/filecoin-project/lotus/pull/11057) to be based off v1.23.3, and get it reviewed.

**Comment by rjan90** [comment]
Closing this ticket now that the SynthPoRep-code has landed in the [`feat/nv21` branch.](https://github.com/filecoin-project/lotus/tree/feat/nv21).

Outstanding item from this epic, that can be seperated into their own tickets as they are more timeline dependent.
- Second round of Butterfly-testing / regression testing after all dependencies landed. Testing plan: https://www.notion.so/pl-strflt/SynthPoRep-Butterfly-network-testing-59e22e4e10a24521b50135571e67391f?pvs=4
- Blog post/media for explaining why all SPs want to use this [luca/irene/nemo]. In draft, but should be published closer to the Calibration-network upgrade IMO

**Comment by cryptonemo** [comment]
>     * Blog post/media for explaining why all SPs want to use this [luca/irene/nemo]. In draft, but should be published closer to the Calibration-network upgrade IMO

FYI, draft is being finalized this week and we are scheduled for blog post release on Oct 3

**Comment by rjan90** [comment]
Awesome @cryptonemo! Perfect timing👌
